test/livebook/proxy_test.exs

```diff
@@ -110,8 +81,23 @@ defmodule LivebookWeb.ProxyPlugTest do
       assert text_response(put(conn, url), 200) == "used PUT method"
       assert text_response(patch(conn, url), 200) == "used PATCH method"
       assert text_response(delete(conn, url), 200) == "used DELETE method"
-
-      Livebook.App.close(pid)
     end
   end
+
+  defp proxy_notebook() do
+    cell =
+      %{
+        Notebook.Cell.new(:code)
+        | source: """
+          Kino.Proxy.listen(fn conn ->
+            conn
+            |> Plug.Conn.put_resp_header("content-type", "application/text;charset=utf-8")
+            |> Plug.Conn.send_resp(200, "used " <> conn.method <> " method")
+          end)\
```

Comment:
Given how small the Kino side is now, I would avoid the dependency and manually follow the same protocol (that's what we do in many other places). Something like this should work:

```suggestion
          fun = fn conn ->
            conn
            |> Plug.Conn.put_resp_header("content-type", "application/text;charset=utf-8")
            |> Plug.Conn.send_resp(200, "used " <> conn.method <> " method")
          end
          
          ref = make_ref()
          send(Process.group_leader(), {:io_request, self(), ref, {:livebook_get_proxy_handler_child_spec, fun}})
          
          child_spec =
            receive do
              {:io_reply, ^ref, {:ok, child_spec}} -> child_spec
            end
          
          start_supervised!(child_spec)\
```

Reply:
`start_supervised!` is not available there, so we can use `Supervisor.start_link(..., ...)` :)

Reply:
It's a separate discussion. I'm not really sure, I kinda want to avoid too much coupling, but making the testing easier is also an argument. I can explore this later :)